src/location.js

```diff
@@ -10,21 +10,45 @@ const Location = ({ homeUrl, children }) => {
   // Set the location state to a parsed url and a compiled url.
   const [parsedUrl, setParsedUrl] = useState(parseJsonApiUrl(homeUrl));
   const [locationUrl, setLocationUrl] = useState(compileJsonApiUrl(parsedUrl));
-  const [document, setDocument] = useState(null);
+  const [responseDocument, setDocument] = useState(null);
+
+  const setUrl = newLocationUrl => {
+    window.history.pushState(
+      {},
+      '',
+      `?location=${encodeURIComponent(newLocationUrl)}`,
```

Comment:
Is it necessary to encode the location query? it is much more readable without being encoded and seems to work.
Pasting a url from the query into the browser also works "unencoded"

Reply:
Hmm, let me try it. I previously was using `encodeURI` (instead of `encodeURIComponent`) and had some bugs. Maybe they'd disappear if I had no encoding at all too.

Reply:
I think we need it. Try setting up some complex includes and sparse fieldsets and then copy the complete URL from the browser bar, open a new tab, paste the URL in and finally visit that URL. You should see that some of the options are lost in the UI (for example, the "Clear `field`" buttons might not appear).

Reply:
I assumed it was a bug with how resources got loaded, but encoding does seem to fix. It would be good if a complex url could be pasted in and have everything load correctly but that could be a future feature.